lib/src/charlatan_response_definition.dart

```diff
@@ -16,20 +16,28 @@ typedef CharlatanRequestMatcher = bool Function(
 /// {@template charlatan_response_builder}
 /// A type representing a function to convert an http request into a response.
 ///
-/// The return value may be any json encodable object, a [CharlatanHttpResponse],
-/// or null.
-///
-/// Typically you will want to return a json response of Map<String, Object?>
-/// which will be automatically serialized and returned with a 200 status code.
-///
-/// If you want to customize the headers or the status code of the response, you
-/// can return an instance of [CharlatanHttpResponse].
+/// The return value is a [CharlatanHttpResponse].
 /// {@endtemplate}
-typedef CharlatanResponseBuilder = FutureOr<Object?> Function(
+typedef CharlatanResponseBuilder = FutureOr<CharlatanHttpResponse> Function(
```

Comment:
While we're at it with a breaking change here, can we just make this a `Future` (and then update the changelog and tests as well)?

```suggestion
typedef CharlatanResponseBuilder = Future<CharlatanHttpResponse> Function(
```

Mainly, I'm thinking about Slava's comment here, about how `FutureOr` is essentially vestigial from the Dart 1 > 2 migration, and that it shouldn't really be necessary: https://twitter.com/mraleph/status/1630322146747510792

Reply:
hm. interesting. the downside to using `Future` here is that it requires all the usages of this typedef to write a buncha extra characters that are pointless if they're not actually implementing anything asynchronously though, right?

i think maybe this is one of the rare cases where we _do_ want a `FutureOr` because then folks can define synchronous fake responses concisely without needing to declare `async` on their lambda when it doesn't do anything async. 

am i missing something there? should i be interpreting that tweet from Slava as if I should be returning `Object` in the typedef and checking `if result is Future<T> { await result } else { result }`? or should i be interpreting it as I should always use `Future` and users should have to make all their lambdas use `async` even when they aren't doing async things?

Reply:
My understanding of the tweet alongside this [SO answer](https://stackoverflow.com/a/59251907) and its comments is that we're using it correctly here. I suspect Slava is getting at the fact that its usage is limited primarily to inputs to things like libraries where the API could conceivably be used either way, and therefore you can offer the flexibility and do the "checking" under the hood (which in theory could probably be added to `Charlatan`, but that's semi-outside of the scope of the thread). What you shouldn't do (as one of the comments in SO points out) is return `FutureOr` from a library function (or any function for that matter) such that the consumer of that function needs to figure out the right way to handle it.

Reply:
Yup all makes sense to me! Thanks both for putting the thought into it 👍🏻

Reply:
Ah yea. Awesome that confirms my thoughts as well and gives me a lot more clarity.

I've pushed up a change to await the value if its a future. I figured that isn't a bad idea here :)